### PR TITLE
Freeze linters

### PR DESCRIPTION
**Pull request by kyessenov**
**What this PR does / why we need it**:
Freezes gometalinter using a docker image.
Removes separate install-prerequisites step for buildifier and gometalinter.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
Vendorization script is adjusted to remove double-symlinks since they don't work well with docker mounts. Generated files are placed into `vendor/` whenever possible for simpler cleanup.

**Release note**:

```release-note
```

### CODE REVIEW

File: bin/check.sh [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+$buildifier -showlog -mode=check \
+  $(find . -type f \( -name 'BUILD' -or -name 'WORKSPACE' -or -wholename '.*bazel$' -or -wholename '.*bzl$' \) -print )
 
 NUM_CPU=$(getconf _NPROCESSORS_ONLN)
 
-gometalinter --concurrency=${NUM_CPU} --enable-gc --deadline=300s --disable-all\
+gometalinter="docker run \
+  -v $(bazel info output_base):$(bazel info output_base) \
+  -v $(pwd):/go/src/istio.io/pilot \
+  gcr.io/istio-testing/linter:bfcc1d6942136fd86eb6f1a6fb328de8398fbd80"
```
Comment (kyessenov) [inline bin/check.sh]: The SHA corresponds to gometalinter SHA. We can automate Dockerfile.lint once we want to get monthly upstream updates.

**Comment by kyessenov** [comment]
cc @ZackButcher @sebastienvas if this looks fine to you, I can do a similar PR in mixer.

File: bin/check.sh [hunk 2]
```diff
@@ -26,10 +36,9 @@ gometalinter --concurrency=${NUM_CPU} --enable-gc --deadline=300s --disable-all\
   --enable=varcheck\
   --enable=vet\
   --enable=vetshadow\
-  ./...
+  /go/src/istio.io/pilot/...
```
Comment (ldemailly) [inline bin/check.sh]: that won't work on people's machine will it ? (my directory is ~/go no /go)

**Comment by kyessenov** [comment]
This refers to a directory inside the docker image.

File: bin/check.sh (hunk 2 above)
Comment (ldemailly) [reply bin/check.sh]: oh right, sorry, I didn't see the docker run on line 11
is that image built each time ? (cached maybe but still) ? 
can we put the linters in some prow image we already have ?

**Comment by kyessenov** [comment]
It's built once. I didn't do automation since linters shouldn't be changing
that often realistically.

File: bin/check.sh (hunk 1 above)
Comment (sebastienvas) [reply bin/check.sh]: Will there be a permission issue here ? Will the link be created by root and not be readable by the user ?

Comment (kyessenov) [reply bin/check.sh]: This image is supposed to be created once a month or even more rarely. Linters don't change that fast. The image should be public readable.

Comment (rshriram) [reply bin/check.sh]: So, this would require docker on the machine in order to even run the linter? That seems like a bummer :(

Comment (rshriram) [reply bin/check.sh]: But if it stabilizes the build infrastructure, so be it.

Comment (kyessenov) [reply bin/check.sh]: i think we can change the script later to use system gometalinter. there's just that line at the end that needs to be changed to `./...` and use `-w` in docker.

**Pull request merged**